apis/config/v1alpha1/license_constraints_types.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+type LicenseConstraints map[string]Restrictions
+
+type Restrictions struct {
+	Constraint string `json:"constraint"`
+	// +optional
+	Distributions []string `json:"distributions,omitempty"`
+}
```

Comment:
How about .. 
```
type LicenseRestrictions map[string]Restriction

type Restriction struct {
	Constraint string `json:"constraint"`
	// +optional
	Distributions []string `json:"distributions,omitempty"`
}
```
LicenseConstraints -> LicenseRestrictions 
Restrictions -> Restriction 

Not avoid mixing "constraint" & "restriction" stufffs